lib/src/metrics/weight-of-class/weight_of_class_metric.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:code_checker/checker.dart';
+
+import '../../utils/declaration_utils.dart';
+
+class WeightOfClassMetric {
+  double compute(
+    ScopedClassDeclaration component,
+    Iterable<ScopedFunctionDeclaration> functions,
+  ) {
+    final totalPublicMethods = componentFunctions(component, functions)
```

Comment:
Seems that u should exclude constructor earlier, because `totalPublicMethods` will always greater than `functionalMethods`